bittensor/_dataset/dataset_impl.py

```diff
@@ -169,46 +175,141 @@ def __del__(self):
     def close(self):
         self.data_queue.close()
 
-    def get_random_directories(self):
-        r""" Getting directories from a random dataset_hash
-        Where a directory could be leading to a data file or a directory file 
+    def get_folder_size(self, folder):
+        total_size = 0
+        full_path = os.path.expanduser(os.path.join(self.data_dir, folder))
+        for dirpath, dirnames, filenames in os.walk(full_path):
+            for f in filenames:
+                fp = os.path.join(dirpath, f)
+                # skip if it is symbolic link
+                if not os.path.islink(fp):
+                    total_size += os.path.getsize(fp)
 
-            Returns:
-                directories (:type:`list`, `required`)
-                    A list of directory.
-                        directory: Map{ Name: str, Hash: str, Size: int }: 
-                            A random directory that lead to a datafile.
+        return total_size
+
+    def load_hash(self, file_meta):
+        r""" Load a hash from disk.
+        Args:
+            file_meta (dict of str: int):
+                Specify the details of the dataset in the format of {'Name': , 'Hash':}.
+
+        Returns:
+            text (str): 
+                The text in the file.                
         """
+
+        full_path = os.path.expanduser(os.path.join(self.data_dir, file_meta['Folder'], file_meta['Hash']))
+        if os.path.exists(full_path):
+            try:
+                with open(full_path, mode='r') as f:
+                    text = f.read()
+
+                logger.success("Loaded from disk:".ljust(20) + "<blue>{}</blue>".format(file_meta['Name']))
+            except Exception:
+                pass
+
+            return text
+        else:
+            return None
```

Comment:
No need for an `else: return None` here, can just do `return None` outside the if statement.